scripts/ci-cd/credential_manager.py

```
@@ -0,0 +1,224 @@
+                # Don't clear it - just warn, as OpenAI may have new formats
+                logging.warning(
+                    "OpenAI API key format may be invalid (doesn't match known patterns)"
+                )
+
+        # Check requirements
+        if self._require_all:
+            missing = [k for k, v in self._credentials.items() if not v]
+            if missing:
+                raise ValueError(f"Missing required credentials: {', '.join(missing)}")
+
+    @lru_cache(maxsize=128)
```

Ruff (0.11.9)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)